interface.go

```diff
@@ -0,0 +1,44 @@
+package idmclient
+
+import (
+	"gopkg.in/macaroon-bakery.v2-unstable/bakery/checkers"
+)
+
+// TODO move these interface declarations into the bakery repository.
+
+// IdentityClient represents an abstract identity manager. A client can
+// use this to create a third party caveat requesting authentication and
+// to decode the authentication information from the resulting discharge
+// macaroon caveats and then to find information about the resulting
+// identity.
+type IdentityClient interface {
```

Comment:
That's something specific to a given identity implementation, I think. In this case, ISTM that the identity server is being used more as a generic third party discharger than an identity service. Specifically the discharge caveat does not contain any declarations in the result, which is what this IdentityClient interface is designed to abstract out. I think that idm.Client should probably implement an IsMemberCaveat method, but currently not convinced that it should be part of IdentityClient.